src/client/controllers/alert-controller.ts

```diff
@@ -0,0 +1,32 @@
+import { Controller } from 'stimulus';
+
+export default class AlertController extends Controller {
+  /**
+   * Dialog element, message, and dismiss button.
+   */
+  public dialogTarget: HTMLElement;
+  public messageTarget: HTMLElement;
+
+  static get targets() {
+    return ['dialog', 'message', 'dismissButton'];
+  }
+
+  public connect() {
+    if (this.data.has('message')) {
+      this.showDialog();
+    }
+    window.alert = (message: string) => {
+      this.data.set('message', message);
+      this.messageTarget.innerText = this.data.get('message');
+      this.dialogTarget.setAttribute('open', '');
+    };
+  }
+
+  public showDialog() {
+    alert(this.data.get('message'));
```

Comment:
I think we should avoid redefining native JavaScript functions if possible, as it's not clear that we're calling something implemented by us not by the browser.  Also, would it be possible to call a private method of `AlertController` here so we don't have to set additional global state?